ARK Server Manager/Lib/ServerRCON.cs

```diff
@@ -370,13 +370,45 @@ private void HandleCommand(ConsoleCommand command)
                     }
                 }).DoNotWait();
             }
+            else
+            {
+                var arkData = await ArkData.ArkDataContainer.CreateRemoteAsync();
+                arkData.SteamLoaded = true;
+                //await arkData.LoadSteamAsync(Config.Default.SteamAPIKey);
+                TaskUtils.RunOnUIThreadAsync(() =>
+                {
+                    foreach (var playerData in arkData.Players)
+                    {
+                        var playerToUpdate = this.Players.FirstOrDefault(p => p.SteamId == Int64.Parse(playerData.SteamId));
+                        if (playerToUpdate != null)
+                        {
+
+                            playerToUpdate.AvatarImage = new BitmapImage(new Uri(playerData.AvatarUrl));
```

Comment:
I understand the reasoning behind the Async part, but at the time was just trying to get the images to load properly.


As far as the SteamLoaded variable goes, I am handling it outside of ArkData because the NodeJS server is actually caching all the steamdata on its side and sending it out with the rest of the player data and if I didn't set the variable to true, ArkData would then try to set all the SteamData itself. I mean, if it doesn't sound very appealing to be manually setting that variable, I could just modify the LoadSteam function?